draft-packages/form/KaizenDraft/Form/Primitives/ToggleSwitch/ToggleSwitch.spec.tsx

```diff
@@ -0,0 +1,38 @@
+import { cleanup, fireEvent, render, screen } from "@testing-library/react"
+import * as React from "react"
+import { ToggledStatus, ToggleSwitchProps } from "./ToggleSwitch"
+import { ToggleSwitch } from "."
+import "@testing-library/jest-dom"
+
+afterEach(cleanup)
+
+const defaultToggleSwitchProps = {
+  id: "someToggleSwitchId",
+  onToggle: jest.fn(),
+  toggledStatus: ToggledStatus.OFF,
+}
+
+const renderToggleSwitch = (props?: ToggleSwitchProps) => {
+  const mergedToggleSwitchProps = { ...defaultToggleSwitchProps, ...props }
+
+  return render(<ToggleSwitch {...mergedToggleSwitchProps} />)
+}
+
+describe("<ToggleSwitch />", () => {
+  it("should call onToggle when toggle is changed", () => {
+    renderToggleSwitch()
+
+    fireEvent.click(screen.getByRole("checkbox"))
+    expect(defaultToggleSwitchProps.onToggle).toHaveBeenCalledTimes(1)
+  })
+
+  it("should show toggledStatus when passed through", async () => {
+    renderToggleSwitch()
+    expect(defaultToggleSwitchProps.toggledStatus).toEqual("off")
```

Comment:
Hmm, is there a way to gather this information from what is rendered instead? I can see that there's `on` and `off` classes that are put on the container element in the implementation. Could that be used?

Reply:
Oh there's classes? Should be able to swing something then. Let me have a poke

Reply:
Yep, that worked a treat. Updated.